src/protocol/ascii.rs

```diff
@@ -95,8 +96,10 @@ impl AsciiProtocol<Stream> {
             return Ok(true);
         } else if s == "NOT_STORED\r\n" {
             return Ok(false);
-        } else if command == StoreCommand::Cas && (s == "EXISTS\r\n" || s == "NOT_FOUND\r\n") {
-            return Ok(false);
+        } else if s == "EXISTS\r\n" {
+            return Err(MemcacheError::from(ResponseStatus::KeyExists as u16));
```

Comment:
Now the binary protocol and ascii protocol do not using a unified error system, I think it should be a issue, and we should resolve it in the future, with some predefined error like `KeyExists` / `KeyNotFound`.

But for now, I think we should keep the ascii protocol using the old (and ugly)  string based error.

Reply:
Didn't get what you're trying to say. We're returning the same error in binary protocol.

Reply:
Just took another look, I thought we should refactor the whole error system, but for now I think it's ok. merging.